src/extension/postgres/extension.rs

```diff
@@ -0,0 +1,287 @@
+    ///
+    /// let mut query = String::new();
+    /// let stmt = ExtensionStatement::create("ltree")
+    ///     .if_not_exists()
+    ///     .to_owned();
+    ///
+    /// stmt.prepare_extension_statement(&mut query);
+    ///
+    /// assert_eq!(query, r#"CREATE EXTENSION IF NOT EXISTS ltree"#);
+    /// ```
+    pub fn if_not_exists(&mut self) -> &mut Self {
+        if matches!(self.operation, ExtensionOperation::Create) {
+            self.if_not_exists = true;
+            return self;
+        }
+
+        panic!("IF NOT EXISTS parameter is not compatible with DROP EXTENSION");
+    }
+
+    /// Uses "IF NOT EXISTS" on Create Extension Statement.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::extension::postgres::{ExtensionBuilder, ExtensionStatement};
+    /// use sea_query::tests_cfg::*;
+    ///
+    /// let mut query = String::new();
+    /// let stmt = ExtensionStatement::create("ltree")
+    ///     .if_not_exists()
+    ///     .to_owned();
+    ///
+    /// stmt.prepare_extension_statement(&mut query);
+    ///
+    /// assert_eq!(query, r#"CREATE EXTENSION IF NOT EXISTS ltree"#);
+    /// ```
+    pub fn if_exists(&mut self) -> &mut Self {
+        if matches!(self.operation, ExtensionOperation::Drop) {
+            self.if_exists = true;
+            return self;
+        }
+
+        panic!("IF EXISTS parameter is not compatible with CREATE EXTENSION");
+    }
+
+    /// Uses "CASCADE" on Create Extension Statement.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::extension::postgres::{ExtensionBuilder, ExtensionStatement};
+    /// use sea_query::tests_cfg::*;
+    ///
+    /// let mut query = String::new();
+    /// let stmt = ExtensionStatement::create("ltree").cascade().to_owned();
+    ///
+    /// stmt.prepare_extension_statement(&mut query);
+    ///
+    /// assert_eq!(query, r#"CREATE EXTENSION ltree CASCADE"#);
+    /// ```
+    pub fn cascade(&mut self) -> &mut Self {
+        self.cascade = true;
+        self
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn create_extension_statement() {
+        let mut writer = String::new();
+        let stmt = ExtensionStatement::create("ltree")
+            .if_not_exists()
+            .cascade()
+            .schema("public")
+            .version("v0.1.0")
+            .to_owned();
+
+        stmt.prepare_extension_statement(&mut writer);
+
+        assert_eq!(
+            writer,
+            r#"CREATE EXTENSION IF NOT EXISTS ltree WITH SCHEMA public VERSION v0.1.0 CASCADE"#
+        );
+    }
+
+    #[test]
+    fn drop_extension_statement() {
+        let mut writer = String::new();
+        let stmt = ExtensionStatement::drop("ltree")
+            .if_exists()
+            .cascade()
+            .to_owned();
+
+        stmt.prepare_extension_statement(&mut writer);
+
+        assert_eq!(writer, r#"DROP EXTENSION IF EXISTS ltree CASCADE"#);
+    }
+}
```

Comment:
Hey @EstebanBorai, thanks for the contribution!! The overall implementation looks great! Just one comment, is it possible to have a similar query building API like the Postgres type statements?

https://github.com/SeaQL/sea-query/blob/9ff95dcfeeec8b99afad1ec6474676c92163c0a6/src/extension/postgres/types.rs#L466-L504

Reply:
Hi @billy1624, to be clear, you refer to implementing the macro?